src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkIndexForAddition(index);
+            Node<T> nextNode = (index == 0) ? first : findNodeByIndex(index);
```

Comment:
Don't create redundant variables. The variable 'nextNode' is unnecessary if you are just using it once in the next line. You can pass the result of 'findNodeByIndex(index)' directly to the 'Node' constructor.